### PR TITLE
Plaid cache

### PR DESCRIPTION
**Pull request by Neitsch**

### CODE REVIEW

File: authentication/plaid_middleware.py [hunk 1]
```diff
@@ -2,18 +2,6 @@
 Plaid setup
```
Comment (infobiac) [inline authentication/plaid_middleware.py]: Given that this no longer creates middleware, perhaps rename plaid_middleware.py to plaid_helper or something?

File: trading/models.py [hunk 2]
```diff
@@ -60,28 +60,30 @@ def has_enough_cash(self, trade_value):
         """
         Check if you have enough cash to make a trade
         """
-        if self.trading_balance() >= trade_value:
+        print("Available:", self.available_cash())
```
Comment (infobiac) [inline trading/models.py]: Remove print statements

File: trading/models.py [hunk 3]
```diff
@@ -92,14 +94,27 @@ def trade_stock(self, stock, quantity):
...
                 stock=stock,
             )
         raise Exception("You don't have the necessary resources!")
 
+    def available_cash(self):
+        """
+        Returns the available cash for the trading account
+        """
+        return (
+            self.trading_balance() +
```
Comment (infobiac) [inline trading/models.py:108]: shouldn't be + trading_balance, should be current_balance() - trading_balance at tradetime.  I think more like: https://github.com/Neitsch/ASE4156/pull/97/files#diff-17490ab817133d204b65862e5d47c947R133

**Review by infobiac — CHANGES_REQUESTED** [review]

File: trading/models.py (hunk 3 above)
Comment (Neitsch) [reply trading/models.py:108]: Discussed offline - it is correct

**Review by infobiac — APPROVED** [review]

**Pull request merged**